lib/msf/base/serializer/opt_conditions.rb

```diff
@@ -0,0 +1,52 @@
+# -*- coding: binary -*-
+module Msf
+  module Serializer
+    module OptConditions
+
+      # Check a condition's result
+      def self.eval_condition(left_value, operator, right_value)
+        result = true
+        case operator
+        when "=="
+          result = left_value == right_value
+        when "!="
+          result = left_value != right_value
+        when "in"
+          result = right_value.include?(left_value)
+        when "nin"
+          result = right_value.include?(left_value)
+          result = !result
+        end
+
+        result
+      end
```

Comment:
A few things about this method. Since the operators are symbolically evaluated, it makes sense to have them as `Symbol`s since they are cheap to compare and create. Ruby's "everything is an expression" approach also allows us to implicitly return the result of the `case` without needing to keep track of an extra variable. Lastly, do we need `==` and `!=` if we could use `in` and `nin` with arrays of one element?

```suggestion
      def self.eval_condition(left_value, operator, right_value)
        case operator
        when :in
          right_value.include?(left_value)
        when :nin
          !right_value.include?(left_value)
        end
      end
```

Reply:
Agree regarding symbols and directly returning the result of the case without the extra variable.

The `==` and `!=` operators still seem necessary though unless there's something I'm missing. The operations aren't the same, and it would be useful for module authors to be able to do explicit comparisons of arrays and scalars via the `==` operator and more loose comparisons of arrays via `in`.

Reply:
Thank you for the feedback. I think the use of == and != is necessary to make it work in this specific way.
In what refers to the use of symbols and the extra variable, I've implemented the suggested changes, plus, changing `operator `to `operator.to_sym`.